loadbalancer/algorithm.go

```diff
@@ -302,46 +186,26 @@ func (ch *consistentHash) Apply(ctx *routing.LBContext) routing.LBEndpoint {
 		return ctx.LBEndpoints[0]
 	}
 
-	choice := ch.chooseConsistentHashEndpoint(ctx, noSkippedEndpoints)
-
-	if ctx.Route.LBFadeInDuration <= 0 {
-		return ctx.LBEndpoints[choice]
-	}
-
-	return withFadeIn(ch.rnd, ctx, choice, ch)
+	choice := ch.chooseConsistentHashEndpoint(ctx)
```

Comment:
It could return negative value (based on the previous version), couldn't it?

Reply:
No, choice could not be negative. `ch.chooseConsistentHashEndpoint` may call `ch.search` or `ch.boundedLoadSearch`, they both end with
```
return ch.hashRing[ringIndex].index
```

All `ch.hastRing[<anything>].index` fields are non-negative, see `newConsistentHashInternal` function which creates this slice:
https://github.com/zalando/skipper/blob/v0.18.23/loadbalancer/algorithm.go#L232

I don't quite understand where this question came up from, maybe it's because of the next line:
```
if ctx.Route.LBFadeInDuration <= 0 {
```
But this if check the completely other value related to fadeIn instead of loadbalancer, that's why I removed it.

Reply:
Ok, got it. Yes, I misinterpreted `if ctx.Route.LBFadeInDuration <= 0 {`